action.js

```diff
@@ -3,40 +3,61 @@ const github = require('@actions/github');
 const fs = require('fs');
 const xmljs = require('xml-js');
 
-let action = async function (name, path, workdirPrefix, githubToken, failOnFailedTests = 'false', failIfNoTests = true) {
-    const {meta, report} = await getReport(path, failIfNoTests);
-
-    let results = `${meta.result}: tests: ${meta.total}, skipped: ${meta.skipped}, failed: ${meta.failed}`;
-    const conclusion = meta.failed === 0 && (meta.total > 0 || !failIfNoTests) ? 'success' : 'failure';
-    core.info(results);
-
-    let annotations = convertReport(report);
-    cleanPaths(annotations, workdirPrefix);
-    await createCheck(githubToken, name, results, failIfNoTests, conclusion, annotations);
-
-    if (failOnFailedTests && conclusion !== 'success') {
-        core.setFailed(`There were ${meta.failed} failed tests`);
+let action = async function (editModepath, playModePath, workdirPrefix, githubToken, failOnFailedTests = false, failIfNoTests = true) {
+    let reports = [
+      {name: 'EditMode Test Results', path: editModepath},
+      {name: 'PlayMode Test Results', path: playModePath}
+    ]; 
+    
+    var i;
+    for (i = 0; i < reports.length; i++) {
+        let {meta, report} = await getReport(reports[i].path, failIfNoTests);
+        if (meta != null && report != null) {
+            processReport(meta, report, reports[i].name, githubToken, workdirPrefix, failOnFailedTests, failIfNoTests);
+        } else {
+            core.info(`No ${reports[i].name} found...`);
+        }
     }
 };
 
 let getReport = async function (path, failIfNoTests) {
+    let meta = null;
+    let report = null;
+    if (!fs.existsSync(path)) {
```

Comment:
I would extract this check to the root function and avoid returning `null` from the function